### PR TITLE
feat: replace hardhat ganache

### PR DESCRIPTION
**Pull request by leoslr**
## Description of the changes

Replace the hardhat-ganache plugin by hardhat:
- Improve smart contract debugging
- Enable better configuration of the chain
- Enable eip1559 calls

Update the eip159 lib and add tests for it.

### CODE REVIEW

File: packages/utils/test/estimate-gas-fee.test.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  });
+
+  /**
+   * The EMA used for estimation occur on the range of 100 blocks.
+   * Over 50 blocks we randomly simulate empty or non-empty blocks.
+   * We check that there's no discrepancy between the results:
+   *  - Less than 20% difference between two consecutive estimations
+   *  - Less than 40% difference between the first and the last estimations
+   */
+  it('Should return a consistent value after several transactions', async () => {
```
Comment (leoslr) [inline packages/utils/test/estimate-gas-fee.test.ts:46]: Open to suggestion to improve this. It's not very practical to test..

**Review by alexandre-abrioux — APPROVED** [review]
thanks 🙂

**Comment by MantisClone** [comment]
Looks good so far. Those erc777 payment processor tests are failing tho :(

File: packages/utils/test/estimate-gas-fee.test.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        baseEstimation.maxFeePerGas as BigNumber,
+        currentEstimation.maxFeePerGas as BigNumber,
+        0.2,
+      );
+      checkEstimation(
+        baseEstimation.maxPriorityFeePerGas as BigNumber,
+        currentEstimation.maxPriorityFeePerGas as BigNumber,
+        0.2,
+      );
+      checkEstimation(
```
Comment (olivier7delf) [inline packages/utils/test/estimate-gas-fee.test.ts]: should we test it only if (i === 49) {...} ?

File: packages/utils/test/estimate-gas-fee.test.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+      firstEstimation.maxFeePerGas?.sub(secondEstimation.maxFeePerGas || 0).toNumber(),
+    ).toBeGreaterThan(0);
+  });
+
+  /**
+   * The EMA used for estimation occur on the range of 100 blocks.
+   * Over 50 blocks we randomly simulate empty or non-empty blocks.
+   * We check that there's no discrepancy between the results:
+   *  - Less than 20% difference between two consecutive estimations
+   *  - Less than 40% difference between the first and the last estimations
```
Comment (olivier7delf) [inline packages/utils/test/estimate-gas-fee.test.ts]:
```suggestion
   *  - Less than 40% difference between first and any other estimate
```

**Review by olivier7delf — APPROVED** [review]

File: packages/utils/test/estimate-gas-fee.test.ts (hunk 3 above)
Comment (leoslr) [reply packages/utils/test/estimate-gas-fee.test.ts]: I think you reviewed this just before I made https://github.com/RequestNetwork/requestNetwork/pull/1072/commits/cc03ae3a26bca3c67e7cdfaef594b16aa074022e 
I think it's better like this:
- No randomization in the test
- The test itself make more sense

**Pull request merged**